app/controllers/mock-login.js

```diff
@@ -11,24 +12,32 @@ export default class MockLoginController extends Controller {
   @service store;
   @tracked isLoading = false;
 
-  @task
-  *queryStore() {
+  queryStore = task(async () => {
     const filter = { provider: 'https://github.com/lblod/mock-login-service' };
     if (this.gemeente) filter.gebruiker = { achternaam: this.gemeente };
-    const accounts = yield this.store.query('account', {
+    const accounts = await this.store.query('account', {
       include: 'gebruiker,gebruiker.bestuurseenheden',
       filter: filter,
       page: { size: this.size, number: this.page },
       sort: 'gebruiker.achternaam',
     });
     return accounts;
-  }
+  });
 
-  @restartableTask
-  *updateSearch(value) {
-    yield timeout(500);
+  updateSearch = task({ restartable: true }, async (value) => {
+    await timeout(500);
     this.page = 0;
     this.gemeente = value;
-    this.model = yield this.queryStore.perform();
+    this.model = await this.queryStore.perform();
+  });
+
+  @action
+  async login(loginF, account) {
+    const accountId = account.id;
+    const gebruiker = account.gebruiker;
```

Comment:
The relationship is async so it needs to be awaited first.
```suggestion
    const gebruiker = await account.gebruiker;
```

Another option is to make it `sync` instead, the (mock) users usually are fetched with includes I think? That would make things a lot simpler.

Reply:
Ah makes sense. I'll make it synchronous. Then we also don't need await and the gebruiker is no longer a proxy which means we can solve the next problem as well.